coral/src/app/features/approvals/acls/AclApprovals.tsx

```diff
@@ -155,7 +156,8 @@ function AclApprovals() {
 
   const { isLoading: rejectIsLoading, mutate: rejectRequest } = useMutation({
     mutationFn: declineAclRequest,
-    onSuccess: (response) => {
+    onSuccess: (responses) => {
+      const response = responses[0];
       if (response.result !== "success") {
         return setErrorMessage(
```

Comment:
Do we need this? That should be covered in the `api.ts` -> `checkForFailureHiddenAsSuccess`in

```
    if (res.result?.toLowerCase().startsWith("failure")) {
      const httpError: HTTPError = {
        data: { message: res.result },
        status: 400,
        statusText: "Bad Request",
        headers: new Headers(),
      };
      return Promise.reject(httpError);
```

As far as I understood backend, result either starts with `success` or `failure`, so in case it's a failure it should never reach the `onSuccess`

Reply:
I believe it's neecessary in this specific case because this endpoint returns an array of results. So the errors are not caught by the block you mentioned. The true fix is of course to handle this case in `api.ts`, though ^^'